frame/benchmarking/src/lib.rs

```diff
@@ -50,1883 +50,226 @@ pub use sp_std::{self, boxed::Box, prelude::Vec, str, vec};
-					results: final_results,
-				});
-			}
-		}
-	)
-}
-
-/// Callback for `define_benchmarks` to call `add_benchmark`.
-#[macro_export]
-macro_rules! cb_add_benchmarks {
-	// anchor
-	( $params:ident, $batches:ident, [ $name:path, $( $location:tt )* ] ) => {
-		$crate::add_benchmark!( $params, $batches, $name, $( $location )* );
-	};
-	// recursion tail
-	( $params:ident, $batches:ident, [ $name:path, $( $location:tt )* ] $([ $names:path, $( $locations:tt )* ])+ ) => {
-		$crate::cb_add_benchmarks!( $params, $batches, [ $name, $( $location )* ] );
-		$crate::cb_add_benchmarks!( $params, $batches, $([ $names, $( $locations )* ])+ );
-	}
-}
-
-/// This macro allows users to easily generate a list of benchmarks for the pallets configured
-/// in the runtime.
 ///
-/// To use this macro, first create a an object to store the list:
-///
-/// ```ignore
-/// let mut list = Vec::<BenchmarkList>::new();
-/// ```
+/// ## Benchmark Tests
 ///
-/// Then pass this `list` to the macro, along with the `extra` boolean, the pallet crate, and
-/// pallet struct:
+/// Benchmark tests can be generated using the old syntax in `frame_benchmarking`,
+/// including the `frame_benchmarking::impl_benchmark_test_suite` macro.
 ///
+/// An example is shown below (taken from the `message-queue` pallet's `benchmarking` module):
 /// ```ignore
-/// list_benchmark!(list, extra, pallet_balances, Balances);
-/// list_benchmark!(list, extra, pallet_session, SessionBench::<Runtime>);
-/// list_benchmark!(list, extra, frame_system, SystemBench::<Runtime>);
+/// #[benchmarks]
+/// mod benchmarks {
+/// 	use super::*;
+/// 	// ...
+/// 	impl_benchmark_test_suite!(
+/// 		MessageQueue,
+/// 		crate::mock::new_test_ext::<crate::integration_test::Test>(),
+/// 		crate::integration_test::Test
+/// 	);
+/// }
 /// ```
-///
-/// This should match what exists with the `add_benchmark!` macro.
-#[macro_export]
-macro_rules! list_benchmark {
-	( $list:ident, $extra:ident, $name:path, $( $location:tt )* ) => (
-		let pallet_string = stringify!($name).as_bytes();
-		let instance_string = stringify!( $( $location )* ).as_bytes();
-		let benchmarks = $( $location )*::benchmarks($extra);
-		let pallet_benchmarks = BenchmarkList {
-			pallet: pallet_string.to_vec(),
-			instance: instance_string.to_vec(),
-			benchmarks: benchmarks.to_vec(),
-		};
-		$list.push(pallet_benchmarks)
-	)
-}
-
-/// Callback for `define_benchmarks` to call `list_benchmark`.
-#[macro_export]
-macro_rules! cb_list_benchmarks {
-	// anchor
-	( $list:ident, $extra:ident, [ $name:path, $( $location:tt )* ] ) => {
-		$crate::list_benchmark!( $list, $extra, $name, $( $location )* );
-	};
-	// recursion tail
-	( $list:ident, $extra:ident, [ $name:path, $( $location:tt )* ] $([ $names:path, $( $locations:tt )* ])+ ) => {
-		$crate::cb_list_benchmarks!( $list, $extra, [ $name, $( $location )* ] );
-		$crate::cb_list_benchmarks!( $list, $extra, $([ $names, $( $locations )* ])+ );
+pub mod v2 {
+	pub use super::*;
```

Comment:
I thought we have some collision here if we import the old stuff?

Reply:
when you do a `use something::*;`, if something local conflicts with something in the use, rust just ignores the conflicting imports, so in this scenario we are importing everything from `frame_benchmarking` _except_ for `benchmarks!` since it conflicts with the local attribute macro `benchmarks`